addons/website_form/static/src/snippets/s_website_form/000.js

```diff
@@ -205,6 +205,10 @@ odoo.define('website_form.s_website_form', function (require) {
                                 if (!currentUrlPath.endsWith("/")) {
                                     currentUrlPath = currentUrlPath + "/";
                                 }
+                                if (!successPage.includes("/#")) {
```

Comment:
This is far from obvious that it actually detects url with anchor having no trailing slash in their path.
At best it looks like a special case for the homepage at first sight.
This if and the previous one could be worth a comment, maybe add a sentence to the existing one "Note that we are adding trailing slash to both URLs for the comparison to match in case of trailing url mismatch" or something like that

Reply:
> This is far from obvious that it actually detects url with anchor having no trailing slash in their path. At best it looks like a special case for the homepage at first sight. This if and the previous one could be worth a comment, maybe add a sentence to the existing one "Note that we are adding trailing slash to both URLs for the comparison to match in case of trailing url mismatch" or something like that

@rdeodoo I made the changes